pyclaragenomics/samples/sample_cudapoa

```diff
@@ -35,41 +35,67 @@ def run_cudapoa(groups, msa, print_output):
     """
     # Get avaialble memory information
     free, total = cuda.cuda_get_mem_info(cuda.cuda_get_device())
+    gpu_mem_per_batch = 0.9 * free
+
+    # Calculate max bounds for sequence size and sequences per POA.
+    max_sequences_per_poa = 0
+    max_seq_sz = 0
+    for group in groups:
+        longest_seq = len(max(group, key=len))
+        max_seq_sz = longest_seq if longest_seq > max_seq_sz else max_seq_sz
+        seq_in_poa = len(group)
+        max_sequences_per_poa = seq_in_poa if seq_in_poa > max_sequences_per_poa else max_sequences_per_poa
+
+    max_sequences_per_poa += 1
```

Comment:
for `max_sequences_per_poa` I guess there is no need for +1. For example in bonito dataset, using 6 is good enough. But for `max_seq_sz` it is definitely needed.

Reply:
Yeah you're right. There's a semantic difference here in the way the two maxes are checked (one is inclusive and the other is not). Let me make the consistent in this PR.

Reply:
It looks like both the max sequence size and max sequences per poa are subject to the same `>=` condition, so I think the `+1` is needed for both cases

https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/dev-v0.5.0/cudapoa/src/cudapoa_batch.hpp#L488

Reply:
A few lines after the comparison for `num_seq`, it gets updated after the comparison: https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/dev-v0.5.0/cudapoa/src/cudapoa_batch.hpp#L506
That's why for `max_sequences_per_poa` +1 is not needed and for the other one it is.
If you change them to be consistent, I think it's better to change them both not to need +1.

Reply:
Cool, thanks for pointing that one out! I've updated the code to make `max_seq_sz` and `max_seqs_per_poa` consistent in upper limit behavior

Reply:
great, thanks.